birdhouse/config/jupyterhub/jupyterhub_config.py.template

```diff
@@ -99,12 +107,21 @@ if readme != "":
 
 def create_dir_hook(spawner):
     username = spawner.user.name
-    user_dir = join(jupyterhub_data_dir, username)
+    jupyterhub_user_dir = join(jupyterhub_data_dir, username)
+
+    if not os.path.exists(jupyterhub_user_dir):
+        os.mkdir(jupyterhub_user_dir, 0o755)
 
-    if not os.path.exists(user_dir):
-        os.mkdir(user_dir, 0o755)
+    subprocess.call(["chown", "-R", "1000:1000", jupyterhub_user_dir])
 
-    subprocess.call(["chown", "-R", "1000:1000", user_dir])
+    if os.environ['WORKSPACE_DIR'] != jupyterhub_data_dir:
+        # Case for cowbird setup. The workspace directory should also have the user's ownership,
+        # to have working volume mounts with the DockerSpawner.
+        workspace_user_dir = join(os.environ['WORKSPACE_DIR'], username)
+        if not os.path.exists(workspace_user_dir):
+            raise FileNotFoundError(f"The user {username}'s workspace doesn't exist in the workspace directory, "
+                                    "but should have been created by Cowbird already.")
+        subprocess.call(["chown", "1000:1000", workspace_user_dir])
```

Comment:
Was removing the `-R` for recursive intentional?
If the directory is not freshly created by Cowbird, could nested contents with different permissions cause problems?

Reply:
Also, currently, when mounting the cowbird user workspace on JupyterHub, the user is able to create notebooks at the root of the `writable-workspace`. I am wondering if there will be confusions between having notebooks in the actual jupyterhub dir (`/writable-workspace/notebooks`, when mounting with cowbird), vs notebooks at the root of `writable-workspace` or other subdirectories.

Reply:
The user should be able to create/modify/delete whatever they want under `notebooks/`, but not under other directories at the same level. Permissions items under those directories would be set individually or recursively (if dirs) based on Magpie permissions.

Reply:
Then I suppose it's alright not to add the `-R` option here, since we don't want to overwrite whatever permission/ownership Magpie would have applied on other directories. 
The `notebooks` directory will also be correct since there is already a `chown` specifically to apply the user `1000` ownership.